Tasks/SonarQubePreBuild/SonarQubePreBuild.ps1

```diff
@@ -21,6 +21,17 @@ Write-Verbose -Verbose "configFile = $configFile"
 Write-Verbose -Verbose "dbConnectionString = $dbUrl"
 Write-Verbose -Verbose "dbUsername = $dbUsername"
 
+$prcaEnabled = Get-TaskVariable -Context $distributedTaskContext -Name "GitPullRequestCodeAnalysisEnabled"
+if ($prcaEnabled -ieq "true")
+{
+    Write-Verbose -Verbose "GitPullRequestCodeAnalysisEnabled is true, setting cmdline args for incremental mode for sonar-runner..."
+
+    if (!$cmdLineArgs.ToString().Contains("sonar.analysis.mode"))
```

Comment:
This doesn't handle the case of sonar.analysis.mode already being set to something other than "incremental".
It might be better to fail if the variable has already been set unless you have a specific scenario that allows the user to choose another analysis mode.